src/ReactFilestack.jsx

```diff
@@ -71,15 +73,16 @@ class ReactFilestack extends Component {
   };
 
   render () {
-    const { buttonClass, buttonText, link } = this.props;
+    const { buttonClass, buttonText, link, children } = this.props;
     const Tag = link ? 'a' : 'button';
     return (
       <Tag
         name="filestack"
         onClick={this.onClickPick}
         className={buttonClass}
       >
-        {buttonText}
+        {!children && buttonText}
```

Comment:
Since you want to go priority to children, what about replacing
```
{!children && buttonText}
{children}
```
with 
```
{children || buttonText}
```

I hope I understood your intentions well.